DbcParserLib/Parsers/ExtendedMultiplexingLineParser.cs

```diff
@@ -0,0 +1,62 @@
+﻿using DbcParserLib.Model;
+using DbcParserLib.Observers;
+using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+
+namespace DbcParserLib.Parsers
+{
+    internal class ExtendedMultiplexingLineParser : ILineParser
+    {
+        private const string SignalLineStarter = "SG_MUL_VAL_ ";
+        private const string SignalRegex = @"SG_MUL_VAL_\s+(\d+)\s+(\S+)\s+(\S+)\s+((?:\d+-\d+,?\s*)+);?";
+
+        private readonly IParseFailureObserver m_observer;
+
+        public ExtendedMultiplexingLineParser(IParseFailureObserver observer)
+        {
+            m_observer = observer;
+        }
+
+        public bool TryParse(string line, IDbcBuilder builder, INextLineProvider nextLineProvider)
+        {
+            if (!line.TrimStart().StartsWith(SignalLineStarter))
+            {
+                return false;
+            }
+
+            var match = Regex.Match(line, SignalRegex);
+
+            if (match.Success)
+            {
+                var multiplexorSignal = match.Groups[3].Value;
+                var multiplexorRanges = match.Groups[4].Value;
+
+                var rangesArray = multiplexorRanges.Split(new[] { ',' }, StringSplitOptions.RemoveEmptyEntries);
+
+                var rangesParsed = new List<MultiplexingRange>();
+
+                foreach (var range in rangesArray)
+                {
+                    var rangeClean = range.Trim();
+                    var numbers = rangeClean.Split('-');
+
+                    var lower = uint.Parse(numbers[0]);
+                    var upper = uint.Parse(numbers[1]);
+
+                    rangesParsed.Add(new MultiplexingRange(lower, upper));
+                }
```

Comment:
Again, just wondering
```cs
var triggerValues = new List<uint>();

foreach (var range in rangesArray)
{
    var rangeClean = range.Trim();
    var numbers = rangeClean.Split('-');

    var lower = uint.Parse(numbers[0]);
    var upper = uint.Parse(numbers[1]);
    
    // Maybe check if upper >= lower?
    rangesParsed.AddRange(Enumerable.Range(lower, upper - lower + 1));
}
```

Reply:
theres so much that could be checked here:
1. Upper is higher or than lower
2. the mux value in the signal definition (m2M) is in the ranges?
3. Is the every mux value possible from the muxor?
4. is the muxor a integer? (or bool which is also an integer in dbc scope)
5. duplicates?

I would go with this:
1. check if upper is higher then lower or else parsing error?
2. use something like a try add to ignore duplicates (not sure atm but ISet is a hashset ? should have a try add just like dictionary)
3. add the value from the signal (m15M) <= 15 in this case too.

 for the final message parsing having ranges sucks anyway it way slower than checking i hashset on contains.